### PR TITLE
Propagate modify time from object to containers

### PR DESCRIPTION
**Pull request by gregorjerse**
Modifications propagate from:
* data -> entity -> collection
* annotation value -> entity
* relation -> collection

REID-2116

### CODE REVIEW

File: resolwe/flow/models/entity.py [hunk 1]
```diff
@@ -377,6 +377,13 @@ def validate_annotations(self):
         if validation_errors:
             raise ValidationError(validation_errors)
 
+    def save(self, *args, **kwargs):
+        """Propagate the modified time to the collection."""
+        super().save(*args, **kwargs)
+        if self.collection:
+            self.collection.modified = self.modified
```
Comment (dblenkus) [inline resolwe/flow/models/entity.py:384]: Does `self.modified` contain updated value?

File: resolwe/flow/models/data.py [hunk 2]
```diff
@@ -570,6 +571,12 @@ def save(self, render_name=False, *args, **kwargs):
 
         with transaction.atomic():
             self._perform_save(*args, **kwargs)
+            if self.entity:
+                self.entity.modified = now()
```
Comment (dblenkus) [inline resolwe/flow/models/data.py:575]: Use `self.modified`.

File: resolwe/flow/models/data.py [hunk 3]
```diff
@@ -570,6 +571,12 @@ def save(self, render_name=False, *args, **kwargs):
...
+                self.entity.save(update_fields=["modified"])
+            elif self.collection:
+                self.collection.modified = now()
```
Comment (dblenkus) [inline resolwe/flow/models/data.py:578]: `self.modified`

File: resolwe/flow/models/entity.py (hunk 1 above)
Comment (gregorjerse) [reply resolwe/flow/models/entity.py:384]: This is a good question. I belive it does, at least according to tests.

File: resolwe/flow/models/data.py (hunk 2 above)
Comment (gregorjerse) [reply resolwe/flow/models/data.py:575]: Done.

File: resolwe/flow/models/data.py (hunk 3 above)
Comment (gregorjerse) [reply resolwe/flow/models/data.py:578]: Done.

**Review by dblenkus — APPROVED** [review]

**Pull request merged**